packages/paste-core/components/heading/__tests__/heading.test.tsx

```diff
@@ -121,38 +110,90 @@ describe('Heading', () => {
             no margin heading
           </Heading>
         </Theme.Provider>
-      )
-      .toJSON();
-    expect(tree).toMatchSnapshot();
+      );
+      expect(asFragment()).toMatchSnapshot();
+    });
   });
 
-  it('should have no accessibility violations', async () => {
-    const container = document.createElement('div');
-    document.body.append(container);
-    render(
-      <Theme.Provider theme="console">
+  describe('HTML attributes', () => {
+    it('should set a element data attribute for Heading', () => {
+      render(
         <Heading as="h1" variant="heading10">
-          This is an H1
-        </Heading>
-        <Heading as="h2" variant="heading20">
-          This is an H2
-        </Heading>
-        <Heading as="h3" variant="heading30">
-          This is an H3
+          heading
         </Heading>
-        <Heading as="h4" variant="heading40">
-          This is an H4
+      );
+      expect(screen.getByRole('heading').getAttribute('data-paste-element')).toEqual('HEADING');
+    });
+    it('should set a custom element data attribute for Heading', () => {
+      render(
+        <Heading as="h1" variant="heading10" element="foo">
+          heading
         </Heading>
-        <Heading as="h5" variant="heading50">
-          This is an H5
-        </Heading>
-        <Heading as="h6" variant="heading60">
-          This is an H6
-        </Heading>
-      </Theme.Provider>,
-      container
-    );
-    const results = await axe(document.body);
-    expect(results).toHaveNoViolations();
+      );
+      expect(screen.getByRole('heading').getAttribute('data-paste-element')).toEqual('foo');
+    });
+  });
+
+  describe('Customization', () => {
+    it('should add custom styles to Heading', (): void => {
+      render(
+        <CustomizationProvider
+          baseTheme="default"
+          elements={{HEADING: {color: 'colorTextWeak', backgroundColor: 'colorBackground'}}}
+        >
+          <Heading as="h1" variant="heading10">
+            Custom heading
+          </Heading>
+        </CustomizationProvider>
+      );
+      const renderedHeading = screen.getByRole('heading');
+      expect(renderedHeading).toHaveStyleRule('background-color', 'rgb(244,244,246)');
+      expect(renderedHeading).toHaveStyleRule('color', 'rgb(96,107,133)');
+    });
+
+    it('should add custom styles to Heading with a custom element data attribute', (): void => {
+      render(
+        <CustomizationProvider
+          baseTheme="default"
+          elements={{foo: {color: 'colorTextWeak', backgroundColor: 'colorBackground'}}}
+        >
+          <Heading as="h1" variant="heading10" element="foo">
+            Custom heading
+          </Heading>
+        </CustomizationProvider>
+      );
+      const renderedHeading = screen.getByRole('heading');
+      expect(renderedHeading).toHaveStyleRule('background-color', 'rgb(244,244,246)');
+      expect(renderedHeading).toHaveStyleRule('color', 'rgb(96,107,133)');
```

Comment:
for any component that has a variant I should be able to target that specific variant to style.

```
elements={{
  HEADING: {
    color: 'colorTextWeak',
    variants: {
      heading10: {
        color: 'colorTextSuccess',
      }
    }
  }
  foo: {
    color: 'colorTextWeak',
    variants: {
      heading10: {
        color: 'colorTextSuccess',
      }
    }
  }
}}
```

I forget the exact syntax, but it's something like that

Reply:
Added stories and tests for customizing a variant. Outside of the `CustomizationProvider` bug, this implementation works.